homeassistant/core.py

```diff
@@ -408,6 +433,8 @@ def async_fire(self, event_type: str, event_data=None,
         for func in listeners:
```

Comment:
I think we talked about this before. At that point in time we decided it would not be a good idea to call it right away because it means callbacks "skip" the line on whatever is waiting in the queue.

Before async, we used to enqueue the jobs based on a priority derived from the event type.